### DISCUSSION TITLE
Cow improvement requests, observations

### DISCUSSION THREAD

**Discussion by jmmullins** [jeremyckahn/farmhand#433] [Ideas]
Improvements / requests:

1. Only an actively selected cow shows a name label on the field? I want to see all the names, please! Also at one point I swear hovering over the unselected cows would show their name labels, but I don't see that happening at the current time
2. It took me a little bit to discover you could rename the cows in your pen. It was mainly an accidental hover that helped me discover it; maybe something about this could be worked into the tip you already have on the homepage about cows? 
3. I would love the concept of being able to add enrichment items / decorations to the pen. They don't need to be moveable everywhere, I think having a few spaces in the pen (like field plots) that allow these "extra items" would be sufficient and players could pick between the available spots to place items. If we wanted them to be more than just decorative, it could help contribute to the cows' happiness stats in some way perhaps. The concept of enrichment items also kind of relates to the "soccer-playing" idea from the [Training thread.](https://github.com/jeremyckahn/farmhand/discussions/182)

Observations/suggestions:

1. The actively selected cow do not move/hop around? I finished hugging the cows for the day and was just watching them, then wondered why one was frozen on the screen while the others hopped around him. When none are selected, they all hop around. I expect this is because you can Hug the cows without having to open the menu by clicking them, but once they have had their max hugs for the day would it be possible to have it deselect focus automatically or allow them to keep hopping? Obviously not critical in any way, just a small thing.
2. When I first bought a pen and got my initial herd, it took me much longer than I'd care to admit to figure out WHERE to buy cow feed from when the initial 25 units ran out. I expected to find it in the normal Shop "supplies" menu alongside the fertilizer and scarecrows, and had to hunt around all the tabs of the main shop, cycled through the different screens, etc. I did find it eventually, but possibly the first time the player runs out of cow feed and it pops up the "oh no, buy some before they get hungry!" notice at the start of the day, on that first time maybe it could be edited to include something like "Buy more cow feed on the Cows page Supplies tab" or something along those lines tacked on to the existing copy.

**Comment by jeremyckahn** [comment]
> 1\. Only an actively selected cow shows a name label on the field? I want to see all the names, please! Also at one point I swear hovering over the unselected cows would show their name labels, but I don't see that happening at the current time

That's a great idea. Right now only selecting a cow will show their name in a tooltip, but I think it would be a nice improvement to show it more often to make the cows more visually distinctive. I'll see what I can do with that soon!

> 2\. It took me a little bit to discover you could rename the cows in your pen. It was mainly an accidental hover that helped me discover it; maybe something about this could be worked into the tip you already have on the homepage about cows?

That's a great idea. It might also be helpful to put something like this icon next to the name field to indicate that it can be changed: https://mui.com/material-ui/material-icons/?query=write&selected=ModeEdit

> 3\. I would love the concept of being able to add enrichment items / decorations to the pen. They don't need to be moveable everywhere, I think having a few spaces in the pen (like field plots) that allow these "extra items" would be sufficient and players could pick between the available spots to place items. If we wanted them to be more than just decorative, it could help contribute to the cows' happiness stats in some way perhaps. The concept of enrichment items also kind of relates to the "soccer-playing" idea from the [Training thread.](https://github.com/jeremyckahn/farmhand/discussions/182)

I love this idea. In general there's a lot of unexplored possibilities for making the cows more interesting within the game. Let's add this to the list! 🙂

> 1\. The actively selected cow do not move/hop around? I finished hugging the cows for the day and was just watching them, then wondered why one was frozen on the screen while the others hopped around him. When none are selected, they all hop around. I expect this is because you can Hug the cows without having to open the menu by clicking them, but once they have had their max hugs for the day would it be possible to have it deselect focus automatically or allow them to keep hopping? Obviously not critical in any way, just a small thing.

This is working as designed, as I figured it would be helpful for players to not have to chase around the cows they're interacting with. That said, I think it would be worth adding in "deselect" functionality. Good thinking!

> 2\. When I first bought a pen and got my initial herd, it took me much longer than I'd care to admit to figure out WHERE to buy cow feed from when the initial 25 units ran out. I expected to find it in the normal Shop "supplies" menu alongside the fertilizer and scarecrows, and had to hunt around all the tabs of the main shop, cycled through the different screens, etc. I did find it eventually, but possibly the first time the player runs out of cow feed and it pops up the "oh no, buy some before they get hungry!" notice at the start of the day, on that first time maybe it could be edited to include something like "Buy more cow feed on the Cows page Supplies tab" or something along those lines tacked on to the existing copy.

Great idea. I can add this soon!
